x-pack/test/api_integration/apis/transform/transforms_preview.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect';
+
+import type { PostTransformsPreviewRequestSchema } from '../../../../plugins/transform/common/api_schemas/transforms';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+import { COMMON_REQUEST_HEADERS } from '../../../functional/services/ml/common_api';
+import { USER } from '../../../functional/services/transform/security_common';
+
+import { generateTransformConfig } from './common';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const transform = getService('transform');
+
+  const expected = {
+    apiTransformTransformsPreview: {
+      previewItemCount: 19,
+      typeOfGeneratedDestIndex: 'object',
+    },
+  };
+
+  function getTransformPreviewConfig() {
+    const { dest, ...config } = generateTransformConfig('the-dummy-id');
```

Comment:
Nit: we could name the transform ids a bit more suite specific, similar to some of the other test files. Maybe something like `transform-test-preview`.

Reply:
Fixed in cd3703a.